lib/FtpContext.js

```diff
@@ -251,7 +252,12 @@ module.exports = class FTPContext {
         // Each response group is passed along individually.
         for (const message of parsed.messages) {
             const code = parseInt(message.substr(0, 3), 10);
-            this._passToHandler({ code, message });
+
+            if (code < 400) {
+                this._passToHandler({ code, message });
+            } else {
+                this._passToHandler(new FTPError(message, {code}));
+            }
```

Comment:
Interesting. This will be one of the key questions: Where/who creates the error object. Until now, I wanted to keep FTPContext as clean as possible from anything too specific to FTP command and return codes. This gives tasks more freedom put also puts a greater burden on them. Not sure about this but good stab for discussion.

Reply:
Yeah, I'm not 100% sure about this, either, but I think makes sense.  It's similar to how the `axios` and (I believe?) `request` modules handle 4XX HTTP responses, so there's precedent for it, if nothing else. :-)